include/openthread/thread_ftd.h

```diff
@@ -195,6 +195,61 @@ otError otThreadSetRouterEligible(otInstance *aInstance, bool aEligible);
  */
 otError otThreadSetPreferredRouterId(otInstance *aInstance, uint8_t aRouterId);
 
+/**
+ * This enumeration represents the power supply configuration on a device.
+ *
+ * This is used as a property in `otDeviceConfig` to calculate the leader weight.
+ *
+ */
+typedef enum
+{
+    OT_POWER_SUPPLY_BATTERY           = 0, ///< Battery powered.
+    OT_POWER_SUPPLY_EXTERNAL          = 1, ///< Externally powered (mains-powered).
+    OT_POWER_SUPPLY_EXTERNAL_STABLE   = 2, ///< Stable external power with a battery backup or UPS.
+    OT_POWER_SUPPLY_EXTERNAL_UNSTABLE = 3, ///< Unstable external power (e.g., a light bulb powered via a switch).
```

Comment:
```suggestion
    OT_POWER_SUPPLY_EXTERNAL_UNSTABLE = 3, ///< Potentially unstable external power (e.g., a light bulb powered via a switch).
```

Reply:
Align better with spec text: it's not that the power is necessarily unstable, but that it may be potentially unstable. (Real stability of it is unknown at manufacture-time, but in some/most homes it could be unstable due to user switching.)